src/strawman/collections/CollectionStrawMan6.scala

```diff
@@ -156,6 +159,8 @@ object CollectionStrawMan6 extends LowPriority {
     }
   }
 
+  type IndexedSeq[+A] = Seq[A] { def view: IndexedView[A] }
```

Comment:
On Sun, Jul 31, 2016 at 12:07 AM, Guillaume Martres <
notifications@github.com> wrote:

> In src/strawman/collections/CollectionStrawMan6.scala
> <https://github.com/lampepfl/dotty/pull/1414#discussion_r72894931>:
>
> > @@ -156,6 +159,8 @@ object CollectionStrawMan6 extends LowPriority {
> >      }
> >    }
> >
> > +  type IndexedSeq[+A] = Seq[A] { def view: IndexedView[A] }
>
> If you do this then any call to view on a value with type IndexedSeq[Foo]
> will end up going through reflection, no?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lampepfl/dotty/pull/1414/files/dabf044485c60245e0fdf64c28eaa90285242a75..8db8c110a2b2a287875222a74511511d80be2b15#r72894931>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAwlVn1LM5xD42dSM8pPrJcmLIQls_Cjks5qa8s5gaJpZM4JUZlv>
> .
>
I don't think so, since `view` is already a method on `Seq`.